src/App.tsx

```diff
@@ -1,109 +1,75 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { useSearchParams } from 'react-router-dom';
+import { Pagination } from './components/Pagination';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
+const items: string[] = getNumbers(1, 42).map(n => `Item ${n}`);
+
+enum ItemsPerPage {
+  three = 3,
+  five = 5,
+  ten = 10,
+  twenty = 20,
+}
 
 export const App: React.FC = () => {
+  const [, setSearchParams] = useSearchParams();
+  const [perPage, setPerPage] = useState(ItemsPerPage.five);
+  const [currentPage, onPageChange] = useState(1);
+
+  const itemsOnPage = items.slice(
+    (currentPage - 1) * perPage,
+    (currentPage - 1) * perPage + perPage,
+  );
+
+  useEffect(() => {
+    setSearchParams({
+      page: currentPage.toString(),
+      perPage: perPage.toString(),
+    });
+  }, [currentPage, perPage, setSearchParams]);
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        Page {currentPage} (items{' '}
+        {itemsOnPage[0].split(' ')[1] && itemsOnPage[0].split(' ')[1]} -{' '}
+        {itemsOnPage[itemsOnPage.length - 1].split(' ')[1] &&
+          itemsOnPage[itemsOnPage.length - 1].split(' ')[1]}{' '}
+        of {items.length})
       </p>
 
-      <div className="form-group row">
-        <div className="col-3 col-sm-2 col-xl-1">
-          <select
-            data-cy="perPageSelector"
-            id="perPageSelector"
-            className="form-control">
-            <option value="3">3</option>
-            <option value="5">5</option>
-            <option value="10">10</option>
-            <option value="20">20</option>
-          </select>
-        </div>
+      <select
+        data-cy="perPageSelector"
+        value={perPage}
+        onChange={e => {
```

Comment:
You are parsing the selected value to a number using the unary plus operator. While this works, it's not immediately clear to someone reading the code. Using `Number(e.target.value)` would make the intention more explicit and the code more readable.